src/finch/compiled.py

```diff
@@ -4,31 +4,54 @@
 from .tensor import Tensor
 
 
-def compiled(func):
-    @wraps(func)
-    def wrapper_func(*args, **kwargs):
-        new_args = []
-        for arg in args:
-            if isinstance(arg, Tensor) and not jl.isa(arg._obj, jl.Finch.LazyTensor):
-                new_args.append(Tensor(jl.Finch.LazyTensor(arg._obj)))
-            else:
-                new_args.append(arg)
-
-        result = func(*new_args, **kwargs)
-        result_tensor = Tensor(jl.Finch.compute(result._obj))
-
-        return result_tensor
-
-    return wrapper_func
-
+def compiled(opt=""):
+    def inner(func):  
+        @wraps(func)  
+        def wrapper_func(*args, **kwargs):  
+            new_args = []  
+            for arg in args:  
+                if isinstance(arg, Tensor) and not jl.isa(arg._obj, jl.Finch.LazyTensor):  
+                    new_args.append(Tensor(jl.Finch.LazyTensor(arg._obj)))  
+                else:  
+                    new_args.append(arg)  
+            result = func(*new_args, **kwargs)  
+            kwargs = {"ctx": get_scheduler(name=opt)} if opt != "" else {}  
+            result_tensor = Tensor(jl.Finch.compute(result._obj, **kwargs))  
+            return result_tensor  
+        return wrapper_func  
+
+    return inner  
 
 def lazy(tensor: Tensor):
     if tensor.is_computed():
         return Tensor(jl.Finch.LazyTensor(tensor._obj))
     return tensor
 
+class AbstractScheduler():
+    pass
+
+class GalleyScheduler(AbstractScheduler):
+    def __init__(self, verbose=False):
+        self.verbose=verbose
+
+class DefaultScheduler(AbstractScheduler):
+    def __init__(self, verbose=False):
+        self.verbose=verbose
+
+def get_julia_scheduler(opt):
+    if isinstance(opt, DefaultScheduler):
+        return jl.Finch.default_scheduler(verbose=opt.verbose)
+    elif isinstance(opt, GalleyScheduler):
+        return jl.Finch.galley_scheduler(verbose=opt.verbose)
+
+def set_optimizer(opt):
+    jl.Finch.set_scheduler_b(get_julia_scheduler(opt))
+    return
 
-def compute(tensor: Tensor, *, verbose: bool = False):
+def compute(tensor: Tensor, *, verbose: bool = False, opt=None, tag=-1):
```

Comment:
I don't know if tag=-1 is respected by Finch anymore, it will just be a tag named "-1" without a sentinel behavior.

Reply:
That's fine I guess. It can just be the default tag, and programs can be cached unless the user makes a new tag.